Tests/SourceKitLSPTests/CodeActionTests.swift

```diff
@@ -1006,6 +1006,64 @@ final class CodeActionTests: XCTestCase {
     }
   }
 
+  func testConvertStringConcatenationToStringInterpolation() async throws {
+    try await assertCodeActions(
+      """
+      1️⃣#"["# + 2️⃣key + ": \\(3️⃣d) " + 4️⃣value + ##"]"##5️⃣
+      """,
+      ranges: [("1️⃣", "2️⃣"), ("3️⃣", "4️⃣"), ("1️⃣", "5️⃣")],
+      exhaustive: false
+    ) { uri, positions in
+      [
+        CodeAction(
+          title: "Convert String Concatenation to String Interpolation",
+          kind: .refactorInline,
+          edit: WorkspaceEdit(
+            changes: [
+              uri: [
+                TextEdit(
+                  range: positions["1️⃣"]..<positions["5️⃣"],
+                  newText: """
+                    ##"[\\##(key): \\##(d) \\##(value)]"##
+                    """
+                )
+              ]
+            ]
+          )
+        )
+      ]
+    }
+  }
+
+  func testConvertStringConcatenationToStringInterpolationNotShowUp() async throws {
+    //    ""
```

Comment:
I think we can simplify this test case to only contain a simple string literals without pounds or string interpolation so that it focuses on one thing and is thus easier to reason about.

---

Leftover comment?